src/AutoMapper/TypeMap.cs

```diff
@@ -72,6 +73,8 @@ public PathMap FindOrCreatePathMapFor(LambdaExpression destinationExpression, Me
         public Type SourceType => SourceTypeDetails.Type;
         public Type DestinationType => DestinationTypeDetails.Type;
 
+        public bool IsReverseMap { get; }
```

Comment:
Maybe have this flag on `MappingExpression` instead?

Reply:
I'm working on this but it needs a bit of refactoring since those objects aren't available everywhere we perform this reverse map creation.

Reply:
Moved this to `ITypeMapConfiguration`